Export robotlist

mathias-luedtke opened the pull request:
addresses #577

mathias-luedtke committed:
use cob_supported_robots_ROBOTLIST in dependent packages

mathias-luedtke commented:
should be used in `cob_calibration_data`as well.

fmessmer commented:
Wow, sweet....that will help keep things consistent quite a lot!

We should do the same for ```ENVLIST``` in ```cob_default_env_config``` and then use ```cob_hardware_config_ROBOTLIST``` and ```cob_default_env_config_ENVLIST``` wherever it is used!

mathias-luedtke commented:
> wherever it is used!

this is tricky as not all robots are configured for high-level packages.

mathias-luedtke committed:
use test_depends where applicable

fmessmer commented:
Sorry, I introduced a merge conflict when merging #578 
But I can take over and try to apply the new cob_hardware_config_ROBOLIST to further repositories....

However, when using this for ```cob_calibration_data```, we are introducing a cyclic dependency...as ```cob_calibration_data``` is used within ```cob_hardware_config```

mathias-luedtke commented:
> Sorry, I introduced a merge conflict

rebased

> However, when using this for cob_calibration_data, we are introducing a cyclic dependency.

New package `cob_supported_robots`?
Or keep list in  `cob_calibration_data`?

fmessmer commented:
> rebased

Also already did that in #582 

 > New package `cob_supported_robots`?
Or keep list in  `cob_calibration_data`?

I would keep the ```robotlist``` in ```cob_calibration_data``` as it is very unlikely that we forget to configure this for a new robot :wink:

mathias-luedtke commented:
> I would keep the robotlist in cob_calibration_data

This even reduces the dependencies (just `xacro`). I wasn't aware that `cob_hardware_config` depends on so many packages.

A `cob_supported_robots`  package would not have any dependencies apart from `catkin`..

floweisshardt commented:
nice solution to mimimise maintenance effort. But I don't like the fact that we introduce dependencies to cob_hardware_config from every package (also simulation related and possibly high-level packages). 

So I vote for introducing a new package `cob_supported_robots`. The question is where to place the package?
- in cob_robots: Bad because of cyclic dependency from/to cob_calibration_data
- as a standalone repository: will solve dependency structure, but will increase amount of overlays
- in cob_calibration_data: good, we just have to make that a multi-package-repo and rename the repository

Alternative would be to use the list out of cob_calibration_data. Pro: no cyclic dependencies because cob_calibration_data has no further dependencies and it doesn't hurt if all packages depend on cob_calibration_data

mathias-luedtke commented:
I vote for keeping the list `cob_calibration_data`, because it is easier to maintain.
A dependency on `xacro`does not hurt that much.

We can migrate to `cob_supported_robots` later if necessary.

> in cob_robots: Bad because of cyclic dependency from/to cob_calibration_data

this is not cyclic!
`cob_robot` -> `cob_supported_robots`
`cob_robot` -> `cob_hardware_config` ->  `cob_calibration_data` ->  `cob_supported_robots`

cyclic (with this PR):
`cob_hardware_config`  -> `cob_calibration_data` ->  `cob_hardware_config`

mathias-luedtke committed:
added cob_supported_robots package

mathias-luedtke commented:
added `cob_supported_robots`

mathias-luedtke commented:
@ipa-fmw: added your changes

fmessmer commented:
Once this is merged we need to update https://github.com/ipa320/cob_simulation/pull/130
Do we want a `cob_supported_environments` package? See https://github.com/ipa320/cob_environments/pull/116
Also, we want to make use of it in `cob_calibration_data`

fmessmer commented:
So, we are going to keep the `robotlist` in [cob_default_robot_behavior](https://github.com/ipa320/cob_robots/blob/indigo_dev/cob_default_robot_behavior/CMakeLists.txt#L8-L14) and [cob_moveit_config](https://github.com/ipa320/cob_robots/blob/indigo_dev/cob_moveit_config/CMakeLists.txt#L8-L17) because those are not implemented for all supported robots, correct?

floweisshardt commented on cob_default_robot_config/package.xml:
```diff
@@ -15,4 +15,6 @@
 
   <depend>roslaunch</depend>
 
+  <build_depend>cob_supported_robots</build_depend>
```
why `build_depend`? other packages have `test_depend` defined

floweisshardt replied:
due to using robotlist in install tags

floweisshardt commented on cob_hardware_config/package.xml:
```diff
@@ -16,6 +16,8 @@
   <depend>roslaunch</depend>
   <depend>rostest</depend>
 
+  <build_depend>cob_supported_robots</build_depend>
```
why `build_depend`? other packages have `test_depend` defined

floweisshardt replied:
due to using robotlist in install tags

fmessmer commented:
`cob_calibration_data` is done in https://github.com/ipa320/cob_calibration_data/pull/115
